src/executor/block_errors.rs

```diff
@@ -0,0 +1,9 @@
+/// Transaction validation errors
+#[derive(thiserror_no_std::Error, Debug, Clone, PartialEq, Eq)]
+pub enum BlockVerificationError {
+    #[error("failed to recover sender for transaction")]
+    SenderRecoveryError,
```

Comment:
Based on the following review comment, we will need to propagate the `SignatureError` (from `ethers`) as well. So we may need something like:

```rust
#[error(transparent)]
SenderRecovery(#[from] ethers_core::types::SignatureError),
#[error("signature verification error: from={0}, recovered={}")]
SenderMismatch(Address, Address)
```

Reply:
But as I mentioned, its best to do this refactoring in a separate PR. I already have a WIP PR for this. So we can keep #19 on hold until we get the refactoring merged.

Reply:
@roynalnaruto can you share your WIP PR?

Reply:
#24